addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -694,6 +694,12 @@ public void inputText(String text) {
         socket.callMethod("Input.SendText", params);
     }
 
+    public void inputExecuteAction(String action) {
```

Comment:
As we have a specific list of possible commands. We should think about a whitelisting the `action` - in the existing `input()` (https://github.com/openhab/openhab2-addons/pull/3123/files#diff-cf13d584f28ada8d7a87e83665490af0R687) method as well. Wdyt?

Reply:
I'm not sure if I catch your drift...

You mean like up, down, left, right, etc. shouldn't occur twice as dedicated Input.[Action] and Input.ExecuteAction([Action] as parameter)?

Reply:
What I wanted to say is that the `action` parameter for the `Input.ExecuteAction` is limited to a fixed enum of possible values. With this binding the user can send every string he likes to this channel. We should maybe filter those commands instead of forwarding them regardless to Kodi.